packages/core-app-api/config.d.ts

```diff
@@ -114,5 +114,11 @@ export interface Config {
      * @visibility frontend
      */
     environment?: string;
+    /**
+     $ The authentication flow type - currently supports 'popup' and 'redirect'
+     * default value: 'popup'
+     * @visibility frontend
+     */
+    authFlow?: string;
```

Comment:
The way this is configured is actually pretty tricky tbh. I think there might be a couple of different levels of config and APIs needed to select which flow is being used. One thing I think would make a lot of sense to support is to have the sign-in page use redirect, but the rest of the app use popup. This means we'd need to be able to mix the flows even for the same provider.

A potential way to do that is to have something like an `enabledAuthFlows: ['popup', 'redirect']` here or something like it, but not sure. Either way I think that long term it's not granular enough to only have a single global switch for this.

One way to avoid having to figure all of this out in this PR is to add an experimental flag instead, and just make it global. For example we'd just make this `enableExperimentalRedirectFlow: true`, and then leave more granular config and a stable API for the future. What do you think?

Reply:
It is totally fine to add a global experimental flag. Would it still be an addition to the config.d.ts interface, just under the "app" parameters? Or is there another approach to adding experimental parameters, and then retrieving the value inside DefaultAuthConnector so that each provider doesn't need to pass in the authFlow value? Are there other examples of experimental flags? 

Thanks.